src/main/java/org/jitsi_modified/impl/neomedia/rtp/remotebitrateestimator/AimdRateControl.java

```diff
@@ -435,11 +439,43 @@ public void update(RateControlInput input, long nowMs)
                 if (input.incomingBitRate > 0L)
                     timeFirstIncomingEstimate = nowMs;
             }
-            else if (nowMs - timeFirstIncomingEstimate > kInitializationTimeMs
-                    && input.incomingBitRate > 0L)
+            else
             {
-                currentBitrateBps = input.incomingBitRate;
-                bitrateIsInitialized = true;
+                long timeSinceFirstIncomingEstimate = nowMs - timeFirstIncomingEstimate;
+                if (timeSinceFirstIncomingEstimate > kFirstIncomingEstimateExpirationMs)
+                {
+                    if (input.incomingBitRate > 0L)
+                    {
+                        timeFirstIncomingEstimate = nowMs;
+                    }
+                    else
+                    {
+                        timeFirstIncomingEstimate = -1L;
+                    }
+
+                    // FIXME Where do we want to track the number of incoming
+                    // bitrate expirations (stored in the incomingBitrateExpirations
+                    // variable below)? The sum of this number across all AIMDs/bridges/deployments
+                    // will be the overall number of times we've prevented the ninjas
+                    // problem thanks to this patch. I see a couple of options:
+                    //
+                    // 1. The debug JSON output feels appropriate but that is not
+                    // something that is accessible/plottable in wavefront afaik.
+                    //
+                    // 2. Next best thing is in the videobridge statistics, but
+                    // it feels a bit awkward to put it there. If we want it in
+                    // there, how should it be bubbled up? event driven or have
+                    // it pulled?
+                    //
+                    // Anything else we should be tracking?
```

Comment:
@bgrozev @JonathanLennox could you please guys take a look and share your thoughts how to implement tracking for this? Maybe we want to push some of the debug stuff to wavefront?

Reply:
The easiest wawy would be to stick it in VideobridgeStatistics -- it already iterates over all endpoints.